.github/workflows/release.yaml

```
@@ -83,12 +83,33 @@ jobs:
       registry-password: ${{ secrets.GITHUB_TOKEN }}
 
   validate:
-    needs: provenance
+    needs: [provenance, build]
     runs-on: ubuntu-latest
     steps:
       - name: Validate image
         uses: enterprise-contract/action-validate-image@latest
         with:
-          image: ${{ env.IMAGE_REGISTRY }}/${{ env.IMAGE_REPO }}:${{ env.DIGEST }}
+          image: ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}
           identity: https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|${{ github.repository_owner }}\/${{ github.event.repository.name }})\/
           issuer: https://token.actions.githubusercontent.com
+    
+  Add-Tag-Validate:
+    runs-on: ubuntu-latest
+    needs: [validate, build]
+    steps:  
+      # needed to login into ghcr.io and push the tag
+      - name: Log in to ghcr
+        uses: redhat-actions/podman-login@9184318aae1ee5034fbfbacc0388acf12669171f # v1
+        with:
+          registry: ${{ env.IMAGE_REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Pull image using Buildah
+        run: buildah pull ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}
+
+      - name: Tag image with latest-validated
+        run: buildah tag ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }} ${{ needs.build.outputs.image }}:latest-validated
```

Thinking about the common use of the `latest` tag, how about we don't tag with `latest` in the build job, perhaps tag with the git id there (`${{ github.sha }}`) and then tag with `latest` once the image has been promoted?

```suggestion
        run: buildah tag ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }} ${{ needs.build.outputs.image }}:latest
```

---

I was going to use the `${{ github. run_id }}` either one IDM